file_example.c

```diff
@@ -97,8 +97,8 @@ file_handler_run(void *arg)
 
 		/* process command */
 		result = file_handle_cmd(h->dev, cmd);
-		pthread_mutex_lock(&state->completion_mtx);
 		tcmulib_command_complete(h->dev, cmd, result);
+		pthread_mutex_lock(&state->completion_mtx);
 		tcmulib_processing_complete(h->dev);
 		pthread_mutex_unlock(&state->completion_mtx);
```

Comment:
Actually if you just have multiple mappings then you delete one mapping while having multiple user:fbo or user:file_async mapped then tcmu-runner will crash.  The odd thing is that the tcmu-runner service will still show active, but if you open up targetcli after the hang then you wont see any tcmu backstores listed, which indicates that tcmu-runner prob crashed 

user:file on the other hand if you have just one mapped and then do a delete it wont show the:

Feb 22 13:04:10 z1403aneo [5778]: remove_device:905 : could not close device fd uio0: 9
Feb 22 13:04:10 z1403aneo [5778]: remove_device:909 : could not unmap device uio0: 22

error, but if you have multiple user:file mapped then delete one mapping it will post that error. 

This is with this patch running, if you dont have this patch running then doing just one mapping and one delete will cause all these problems....